study-builder/fdahpStudyDesigner/src/main/webapp/js/common.js

```diff
@@ -117,6 +117,39 @@ $(document)
                     }
 
                   });
+          $('input[type = text][custAttType = customValidate]')
+              .on(
+                  'keyup',
+                  function (e) {
+                    var evt = (e) ? e : window.event;
+                    var charCode = (evt.which) ? evt.which
+                        : evt.keyCode;
+                    if (charCode == 16)
+                      isShift = false;
+                    if (!isShift && $(this).val()) {
+                      var regularExpression = /^[a-z0-9_-]*$/;
+                      if (!regularExpression.test($(this)
+                          .val())) {
+                        var newVal = $(this)
+                            .val()
+                            .replace(
+                                /[^a-z0-9_-]/g,
+                                '');
+                        e.preventDefault();
+                        $(this).val(newVal);
+                        $(this).parent().addClass(
+                            "has-danger has-error");
+                        $(this)
+                            .parent()
+                            .find(".help-block")
+                            .empty()
+                            .append($("<ul><li> </li></ul>")
+                            .attr("class","list-unstyled")
+                            .attr("style","white-space:nowrap")
+                            .text("Please use allowed characters only: lowercase alphabets (a-z), digits (0-9), _ (underscore) and -(minus)."));
```

Comment:
Alphabets is not the correct term for this - get tech writer help

Reply:
Fixed review comment